document/CoreFormat.mdk

```diff
@@ -1120,11 +1185,78 @@ C++: `Namespace::Class::Method(int, double) const &&`
 C#: `Namespace1.Namespace2.Class.Method(System.String, int[])`
 ~
 
+If the `run.logicalLocations` property (&sect;[#run-logicalLocations]) is present,
+the value of the `fullyQualifiedLogicalName` property should be equal to the name of one of the properties on the `run.logicalLocations` object,
+with one exception, described in &sect;[#location-logicalLocationKey].
+
 ~ IsoNote
 The intent of this property is to allow a result log viewer to display the logical location
```

Comment:
Ok, I've thought about this some more. Those three rationales (1. fingerprinting, 2. analysis without PDB information, and 3. human readability) are really addressing two different issues:

1. Why include logical location information in SARIF at all, when you already have physical location information?

2. Why include the `fullyQualifiedLogicalName` property, when all the information in it is captured in the array of `logicLocationComponent` objects?

The sentence you commented on -- and remember, it was written at a time when `fullyQualifiedLogicalName` and the `logicalLocationComponent` array were adjacent to each other, both children of `location` -- was intended to answer #2, "why have `fullyQualifiedLogicalName` when you already have the more detailed `logicalLocationComponent`?" The answer was "to help viewers provide the logical information in a readable way".

On the other hand, the two rationales you cited address #1, "why have both physical _and_ logical locations?". And the answers are "because logical information is more resilient, hence better for fingerprinting", and "because it's sometimes available when source information is not".

(The argument from resilience justifies logical location information in general, not the fQLN property in particular. You could just as well fingerprint the `logicalLocationComponent` array.)

Therefore I think it's appropriate to split those two concerns in the spec. I'll address #1, with your two rationales, in the "location - General" section. I'll leave the answer to #2 where it is, in the "location - fullyQualifiedLogicalName" section. But I'll be clearer about which of the two concerns each one is addressing.